nibe/connection/modbus.py

```diff
@@ -24,6 +25,17 @@ def split_modbus_data(coil: Coil):
     return entity_type, entity_address, entity_count
 
 
+def split_chunks(data, max_len, chunks) -> List[bytes]:
+    """Split data into chunks of a certain max length, cropping of trailing data."""
+    count = len(data) // chunks
+    res = []
+    for i in range(0, len(data), count):
+        chunk = data[i : i + count]
+        assert all(x == 0 for x in chunk[max_len:])
+        res.append(chunk[0:max_len])
+    return res
+
+
```

Comment:
This is rather silly. I would like to use Construct here to handle the packing of data. So we can pack a uint32 into two 16bit blocks and and a list of uint32 into a list of uint16.

Would also be nice to handle packing a list of bits, into bits packed into uint8's. To get this to work, we would need to move the raw/non_raw parsing away from coil.py to the connection instead.

Reply:
I was thinking that it would hit us later. I think I can do it in upcoming days.

Reply:
I have moved it to a separate class.